### PR TITLE
DDF-1343: Fixed LDAP filter to allow for appending to factoryPid

### PR DESCRIPTION
**Pull request by AzGoalie**
Allows for factoryPids that don't end in Source or Service
@mverret Will hero
@shaundmorris 
@brendan-hofmann

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-catalog/136)

### CODE REVIEW

**Comment by brendan-hofmann** [comment]
:+1:  but needs documentation updates.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

<sup>**[admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java, line 247 [r1]](https://reviewable.io:443/reviews/codice/ddf-catalog/136#-JvyDjSpeYEZVjyGw0hc)** ([raw file](https://github.com/codice/ddf-catalog/blob/7ca52b68efa968462da6358296f6087f4a4f99b7/admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java#L247)):</sup>
Aha, this is what we missed in the initial review, we went from a .contains() on the UI side to a .endsWith() on the backend, effectively.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/136)

**Comment by shaundmorris** [comment]
LGTM. I am wondering if we should document some guidance on Source Factory PIDs, but we don't create many types of those very often. 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/136)

**Comment by kcwire** [comment]
I think with this approach we need to have standards on how we name any PID.  There is nothing enforcing me to create a new FederatedSource and include the word `source` or `service` in it's PID.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/136)

**Comment by brendan-hofmann** [comment]
@kcwire I talked to @AzGoalie about updating the documentation. I'm not sure how else this would be enforced, unless you want the catalog framework to check for proper naming on sources and throw warnings if they're not named correctly.

File: docs/src/main/resources/ExtContents.adoc [hunk 1]
```diff
@@ -2912,12 +2912,16 @@ _Similar to a Federated Source with the following exceptions:_
...
-. Create a new class that implements the specified Source interface.
+. Create a new class that implements the specified Source interface and `ConfiguredService`.
 . Implement the required methods.
 . Create an OSGi descriptor file to communicate with the OSGi registry. (Refer to the OSGi Services  section.)
 .. Import DDF packages.
 .. Register source class as service to the OSGi registry.
 . Deploy to DDF. (Refer to the Working with OSGi - Bundles section.)
+[IMPORTANT]
+====
+The `factory-pid` property of the metatype must contain either service or source in the name
```
Comment (brendan-hofmann) [inline docs/src/main/resources/ExtContents.adoc]: The filter as written is a little more strict than that, isn't it? Unless the ldap filter is evaluated case-insensitively, the actual requirement is that the factory-pid must contain one of "Source|source|Service|service", literally.

If the ldap filter is actually case-insensitive, you could also simplify the filter query, naturally. ;)

**Comment by AzGoalie** [comment]
Closing this pr since the repos got merged. New pr is here https://github.com/codice/ddf/pull/92

**Pull request closed without merging**